dataset_loaders/videos/davis2017.py

```diff
@@ -0,0 +1,279 @@
+from __future__ import division
+import os
+import time
+
+import numpy as np
+from PIL import Image
+
+from dataset_loaders.parallel_loader import ThreadedDataset
+from dataset_loaders.utils_parallel_loader import natural_keys
+
+floatX = 'float32'
+
+
+class Davis2017Dataset(ThreadedDataset):
+    name = 'davis2017'
+
+    non_void_nclasses = None
+    _void_labels = []
+
+    # NOTE: we only load the 480p
+    # 1080p images are either (1920, 1080) or (1600, 900)
+    data_shape = (480, 854, 3)
+    # _cmap = {
+    #     0: (255, 255, 255),        # background
+    #     1: (0, 0, 0)}              # foreground
+    # _mask_labels = {0: 'background', 1: 'foreground'}
+
+    _unique_rgbs = None
+    _filenames = None
+    _image_sets_path = None
+    _prefix_list = None
+
+    @property
+    def prefix_list(self):
+        if self._prefix_list is None:
+            # Create a list of prefix out of the number of requested videos
+            with open(os.path.join(self._image_sets_path,
+                                   self.which_set + '.txt')) as f:
```

Comment:
Using 'test-dev' in which_set is a poor way to get this line to work. replace with
```python
which_set = 'test-dev' if self.which_set == 'test' else self.which_set
with open(os.path.join(self._image_sets_path,
                       which_set + '.txt')) as f: